src/compiler/sys.ts

```diff
@@ -541,11 +541,20 @@ namespace ts {
         // patch writefile to create folder before writing the file
         const originalWriteFile = sys.writeFile;
         sys.writeFile = (path, data, writeBom) => {
-            const directoryPath = getDirectoryPath(normalizeSlashes(path));
-            if (directoryPath && !sys.directoryExists(directoryPath)) {
-                recursiveCreateDirectory(directoryPath, sys);
+            // PERF: Checking for directory existence is expensive.

```

Comment:
Can you please experiment moving this arrow in to helper function that takes host with `writeFile`, `createDirectory` and `directoryExists` so that our all implementations use that helper instead of three different places copying same thing in some different way.

Reply:
The three directory existence checks were inconsistent, but I couldn't see a reason for the inconsistency, so I picked what seemed like a sensible implementation for the shared helper.  In tsc, two of the call-sites will never trigger directory creation anyway.